pkg/imagestore/builder.go

```diff
@@ -0,0 +1,17 @@
+package imagestore
+
+import (
+	"io"
+)
+
+// Builder is a means of creating image stores.
+type Builder interface {
+	// ArchiveDir creates a fresh Builder with the given archive directory.
+	ArchiveDir(string) Builder
+
+	// Logs creates a fresh builder with the given log stream.
```

Comment:
This is rather unintuitive - maybe `NewBuilder`?

Reply:
Maybe the description is confusing, but this is just the builder pattern which allows chaining. It doesn't mutate the "receiver" but rather returns a fresh builder with the logs value set. Not sure how to explain this better in the code without linking to the builder pattern somewhere...

Reply:
Ok,I thought that might be the case - maybe consider a more intuitive naming for the method, like `SetLogs`?

The only reservation I have around the naming for this is the fact that we have an additional `Builder` interface which also has a `Logs()` method, but that one returns the logs from the build - which might generate confusion.
But if you have strong opinions about that, we can also have a look and consider refactoring later.